scheduled/cru-org-feed.js

```diff
@@ -0,0 +1,162 @@
+'use strict'
+
+const rollbar = require('../config/rollbar')
+const {Client} = require('pg')
+const Transform = require('stream').Transform
+const QueryStream = require('pg-query-stream')
+const CSV = require('csv')
+const AWS = require('aws-sdk')
+const request = require('request-promise-native')
+const {find, map, zipObject} = require('lodash')
+const prefixMatch = new RegExp('^https?://(www\\.)?cru\\.org(/content/cru)?', 'i')
+const feedColumns = [
+  'entity.id',
+  'entity.name',
+  'entity.categoryId',
+  'entity.message',
+  'entity.thumbnailUrl',
+  'entity.value',
+  'entity.pageUrl',
+  'entity.inventory',
+  'entity.margin',
+  'entity.content_score',
+  'entity.language',
+  'entity.primary_category',
+  'entity.sub_categories'
+]
+const csvHeader = `## RECSRecommendations Upload File
+## RECS''## RECS'' indicates a Recommendations pre-process header. Please do not remove these lines.
+## RECS
+## RECSUse this file to upload product display information to Recommendations. Each product has its own row. Each line must contain 19 values and if not all are filled a space should be left.
+## RECSThe last 100 columns (entity.custom1 - entity.custom100) are custom. The name 'customN' can be replaced with a custom name such as 'onSale' or 'brand'.
+## RECSIf the products already exist in Recommendations then changes uploaded here will override the data in Recommendations. Any new attributes entered here will be added to the product''s entry in Recommendations.
+## RECS${feedColumns.join(',')}
+`
+const scoresQuery = `SELECT MD5(lower(uri)) AS "entity.id", uri AS "entity.pageUrl", score AS "entity.content_score" FROM scores WHERE uri ~* $1`
+
+module.exports.handler = async (lambdaEvent) => {
+  try {
+    // Configure AWS and S3
+    AWS.config.update({region: 'us-east-1'})
+    const s3 = new AWS.S3({apiVersion: '2006-03-01'})
+
+    /**
+     * Fetches cru.org json information through querybuilder
+     * @returns {Promise<void>}
+     */
+    const getCruOrgJson = async () => {
+      return request({
+        method: 'GET',
+        url: 'http://prodpub1.aws.cru.org:4503/bin/querybuilder.json',
```

Comment:
This won't work if pub1 is down, but pub2 is up. Not sure if there is a loadbalanced URL for the publisher directly. If not maybe we could do a retry (if pub1 fails, try pub2).